codalab/apps/web/views.py

```diff
@@ -1949,6 +1949,138 @@ def get_success_url(self):
         return reverse('competitions:dumps', kwargs={'competition_pk': dump.competition.pk})
 
 
+class CompetitionWidgetsView(LoginRequiredMixin, DetailView):
+    queryset = models.Competition.objects.all()
+    template_name = 'web/competitions/widgets.html'
+
+    def get_object(self, queryset=None):
+        self.object = super(CompetitionWidgetsView, self).get_object()
+        if self.request.user != self.object.creator and self.request.user not in self.object.admins.all():
+            raise Http404()
+        return self.object
+
+
+class WidgetMixin(object):
+
+    def get_context_data(self, **kwargs):
+        context = super(WidgetMixin, self).get_context_data(**kwargs)
+        context['current_phase'] = get_current_phase(self.object)
+        return context
+
+
+class CompetitionSubmissionWidgetView(WidgetMixin, DetailView):
+    queryset = models.Competition.objects.all()
+    template_name = 'web/widget_iframes/submission.html'
+
+    def get_context_data(self, **kwargs):
+        context = super(CompetitionSubmissionWidgetView, self).get_context_data(**kwargs)
+        context['phase'] = get_current_phase(self.object)
+        competition = self.object
+
+        if settings.USE_AWS:
+            context['form'] = forms.SubmissionS3UploadForm
+
+        if competition.participants.filter(user__in=[self.request.user]).exists():
+            participant = competition.participants.get(user=self.request.user)
+            if participant.status.codename == models.ParticipantStatus.APPROVED:
+                phase = context['phase']
+
+                submissions = models.CompetitionSubmission.objects.filter(
+                    participant=participant,
+                    phase=phase
+                ).select_related('status').order_by('submitted_at')
+
+                # find which submission is in the leaderboard, if any and only if phase allows seeing results.
+                id_of_submission_in_leaderboard = -1
+                if not phase.is_blind:
+                    leaderboard_entry = models.PhaseLeaderBoardEntry.objects.filter(
+                        board__phase=phase,
+                        result__participant__user=self.request.user
+                    ).select_related('result', 'result__participant')
+                    if leaderboard_entry:
+                        id_of_submission_in_leaderboard = leaderboard_entry[0].result.pk
+                submission_info_list = []
+                for submission in submissions:
+                    try:
+                        default_score = float(submission.get_default_score())
+                    except (TypeError, ValueError):
+                        default_score = '---'
+                    submission_info = {
```

Comment:
Should this be surrounded in a Try/Except as well in case one of these fields on submission is None? Shouldn't really matter but what do we have for error checking on the values?

Reply:
I basically just snatched this code out of the other view, I believe it should handle problems fine. Biggest downside is it's not very DRY but it should be the same as we have currently